flocker/common/_filepath.py

```diff
@@ -38,3 +42,55 @@ def make_directory(path):
     if not path.isdir():
         path.makedirs()
     return path
+
+
+class IFilePathExtended(IFilePath):
+    """
+    ``IFilePath`` plus some other ``FilePath`` methods that we need in Flocker.
+    """
+    def remove():
```

Comment:
It's a `zope.interface` thing. 
TBH I never really understood why until now.
The docs have this to say:
> Interfaces document how an object is used. When calling instance methods, you don’t pass a self argument, so a self argument isn’t included in the interface signature. The self argument in instance methods is really an implementation detail of Python instances. Other objects, besides instances can provide interfaces and their methods might not be instance methods. For example, modules can provide interfaces and their methods are usually just functions. Even instances can have methods that are not instance methods.

https://docs.zope.org/zope.interface/README.html